ui/src/main/java/edu/wpi/grip/ui/preview/PreviewsView.java

```diff
@@ -48,24 +58,196 @@ public PreviewsView(EventBus eventBus) {
     }
 
     @Subscribe
+    /**
+     * This function is called when a step moves in the pipeline to adjust the positions of any open previews it has
+     * to reflect the new order of the pipeline.
+     */
+    public synchronized void onPreviewOrderChanged(StepMovedEvent event) {
```

Comment:
So, just curious, but is there a reason why you didn't use some pre-defined sorting function?  Seems like it would be nice to not have all of this manual keeping track of positions and stacks and stuff.

    ObservableList<SocketPreviewView<?>> previews = FXCollections.observableArrayList(
            (Collection) previewBox.getChildren());

    Collections.sort(previews, (a, b) -> {
        if (...) {
            return -1; // a goes before b
        } else {
            return -1; // b goes before a
        }
    });

    previewBox.getChildren().setAll(previews);